heft-plugins/heft-sass-plugin/src/SassTypingsGenerator.ts

```diff
@@ -133,45 +181,43 @@ export class SassTypingsGenerator extends StringValuesTypingsGenerator {
     buildFolder: string,
     importIncludePaths: string[] | undefined
   ): Promise<string> {
-    const result: Result = await LegacyAdapters.convertCallbackToPromise(render, {
-      data: fileContents,
-      file: filePath,
-      importer: (url: string) => ({ file: this._patchSassUrl(url) }),
-      includePaths: importIncludePaths
-        ? importIncludePaths
-        : [path.join(buildFolder, 'node_modules'), path.join(buildFolder, 'src')],
-      indentedSyntax: path.extname(filePath).toLowerCase() === '.sass'
-    });
+    const result: Result = await new Promise(
```

Comment:
This style is perfectly fine with me, but I just want to make sure we don't have a particular style guideline for callback apis. @iclanton or @D4N14L do we recommend using LegacyAdapters.convertCallbackToPromise wrapped in try/catch, or is creating a new promise acceptable or perhaps preferred?

Reply:
Hmm, in all honesty I kinda like using `Promise` directly when the signature is more complicated, it seems to clarify to me what is going on. However, it's also very doable to use the `LegacyAdapters` method and catch the resulting error. The original error should bubble up anyway, so this seems like a stylistic preference.

Reply:
I did this because I wanted to be clearer about that the error type was a SassError object, while Promise makes no guarantees about the shape of its rejected error type.

Reply:
The SassError contains the extra file, column, and line information which is very helpful to developers when surfacing the error.

Reply:
Sounds good to me!